### PR TITLE
Add tappable fields in Overview cards

### PR DESCRIPTION
**Pull request by Stereopoly**
* Please test! (on actual devices too, since no Phone on sim)

Simulator Screen Shot - iPhone 11 Pro - 2020-10-03 at 16 17 44
Simulator Screen Shot - iPhone 11 Pro - 2020-10-03 at 16 17 52

### CODE REVIEW

File: berkeley-mobile/Utils/UIViewController+Extensions.swift [hunk 1]
```diff
@@ -38,4 +40,42 @@ extension UIViewController {
...
+    public func presentAlertLinkTelephone(title: String, message: String, options: String..., phoneNumber: String) {
+        let alertController = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        guard let number = URL(string: "tel://\(phoneNumber)") else { return }
+        
+        for (index, option) in options.enumerated() {
+            if (index == 0) {
+                alertController.addAction(UIAlertAction.init(title: option, style: .cancel))
+            } else {
+                alertController.addAction(UIAlertAction.init(title: option, style: .default, handler: { _ in
+                    UIApplication.shared.open(number, options: [:])
```
Comment (KevWho) [inline berkeley-mobile/Utils/UIViewController+Extensions.swift:54]: Wonder if you should add a check using `UIApplication.shared.canOpenURL(_:)` before showing the alert or display some sort of response on failure. As is, if the number cannot be dialed (e.g. the user is on an iPad w/o a SIM or something) nothing happens after the user presses "Yes", which might be a bit confusing.

File: berkeley-mobile/Common/DetailView/OverviewCardView.swift [hunk 2]
```diff
@@ -172,6 +186,26 @@ class OverviewCardView: CardView {
...
+            UIApplication.shared.windows.first!.rootViewController!.presentAlertLinkTelephone(title: "Are you sure you want to open Phone?", message: "Berkeley Mobile wants to call \(gesture.phoneNumber)", options: "Cancel", "Yes", phoneNumber: gesture.phoneNumber.trimmingCharacters(in: .whitespacesAndNewlines).replacingOccurrences(of: "-", with: ""))
+        }
+    }
+    
+    @objc func addressTapped(gesture: DetailTapGestureRecognizer) {
+        if (gesture.view as? UILabel) != nil {
+            // Kinda hacky - necessary since cannot present alert when Popover is visible (ex. on Resources page)
+            self.window!.rootViewController?.dismiss(animated: true, completion: nil)
+            
+            UIApplication.shared.windows.first!.rootViewController!.presentAlertLinkMaps(title: "Are you sure you want to open Maps?", message: "Berkeley Mobile wants to navigate to \(self.addressLabel.text!)", options: "Cancel", "Yes", lat: gesture.latitude, lon: gesture.longitude, name: self.addressLabel.text!)
```
Comment (KevWho) [inline berkeley-mobile/Common/DetailView/OverviewCardView.swift:204]: Some items have addresses but no longitude / latitude (e.g. Suicide Resource Prevention Center). Tapping the address for these items opens Maps to the middle of the ocean (0, 0). Maybe you could make all the properties (or just lat / lon) in `DetailTapGestureRecognizer` optional and do some quick nil checks at the top of these functions?

**Review by KevWho — CHANGES_REQUESTED** [review]
Nice, works great on my device. Requested a few changes.

File: berkeley-mobile/Common/DetailView/OverviewCardView.swift [hunk 3]
```diff
@@ -172,6 +186,26 @@ class OverviewCardView: CardView {
         }
     }
     
+    //MARK: - Label Selectors
+    @objc func phoneNumberTapped(gesture: DetailTapGestureRecognizer) {
+        if (gesture.view as? UILabel) != nil {
+            // Kinda hacky - necessary since cannot present alert when Popover is visible (ex. on Resources page)
+            self.window!.rootViewController?.dismiss(animated: true, completion: nil)
+            
+            UIApplication.shared.windows.first!.rootViewController!.presentAlertLinkTelephone(title: "Are you sure you want to open Phone?", message: "Berkeley Mobile wants to call \(gesture.phoneNumber)", options: "Cancel", "Yes", phoneNumber: gesture.phoneNumber.trimmingCharacters(in: .whitespacesAndNewlines).replacingOccurrences(of: "-", with: ""))
```
[the remark this answers is not in the document]
Comment (Stereopoly) [reply berkeley-mobile/Common/DetailView/OverviewCardView.swift:195]: Ooh ya def don't need an alert then

**Review by KevWho — APPROVED** [review]
Mapping works! I'm sure we might discover some edge cases in the future where it does something unexpected, though, haha

**Pull request merged**